frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetailsActions.tsx

```diff
@@ -18,38 +21,57 @@ const PipelineDetailsActions: React.FC<PipelineDetailsActionsProps> = ({ onDelet
   const navigate = useNavigate();
   const { namespace } = usePipelinesAPI();
   const [open, setOpen] = React.useState(false);
+  const [isVersionImportModalOpen, setIsVersionImportModalOpen] = React.useState(false);
 
   return (
-    <Dropdown
-      onSelect={() => setOpen(false)}
-      toggle={
-        <DropdownToggle toggleVariant="primary" onToggle={() => setOpen(!open)}>
-          Actions
-        </DropdownToggle>
-      }
-      isOpen={open}
-      position="right"
-      dropdownItems={[
-        // TODO: Handle path
-        <DropdownItem
-          key="create-run"
-          onClick={() =>
-            navigate(`/pipelineRuns/${namespace}/pipelineRun/create`, {
-              state: { lastPipeline: pipeline },
-            })
+    <>
+      <Dropdown
+        onSelect={() => setOpen(false)}
+        toggle={
+          <DropdownToggle toggleVariant="primary" onToggle={() => setOpen(!open)}>
+            Actions
+          </DropdownToggle>
+        }
+        isOpen={open}
+        position="right"
+        dropdownItems={[
+          <DropdownItem key="upload-version" onClick={() => setIsVersionImportModalOpen(true)}>
+            Upload new version
+          </DropdownItem>,
+          <DropdownSeparator key="separator-1" />,
+          // TODO: Handle path
```

Comment:
Are you going to solve this in this PR? Or you are going to leave it somewhere else? When I click the create run button, the pipeline is auto-selected but the version is selected as the default version instead of the current view version selection in the pipeline details page.

Reply:
I didn't write that TODO comment, that was there before this PR. Both of the "run" actions weren't mentioned for this ticket.

Reply:
I've updated the state and handling of that state to include the "lastVersion" to prefill the create run page with the version from the pipeline details page.